src/transformers/models/whisper/generation_whisper.py

```diff
@@ -474,11 +473,8 @@ def generate(
                 "The input name `inputs` is deprecated. Please make sure to use `input_features` instead.",
                 FutureWarning,
             )
-        # 1. copy generation config
-        if generation_config is None:
-            generation_config = copy.deepcopy(self.generation_config)
-        else:
-            generation_config = copy.deepcopy(generation_config)
+        # 1. prepare generation config
+        generation_config, kwargs = self._prepare_generation_config(generation_config, **kwargs)
```

Comment:
The lines above imply there's a `self.generation_config` which should be used if `generation_config` is `None`

Reply:
`_prepare_generation_config` does precisely that: https://github.com/huggingface/transformers/blob/484e10f7f26b62452ba96488811a0c3f6eaf58df/src/transformers/generation/utils.py#L1204

It is a more complex version of this original if/else that preserves additional backward (and forward!) compatibility features of `generate` :)